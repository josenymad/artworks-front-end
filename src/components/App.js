import React, { useEffect, useState } from "react";
import "../styles/app.css";
import { Route, Routes } from "react-router-dom";
import certagsLogo from "../images/certags-web.png";
import NavBar from "./NavBar";
import AllArtworks from "./AllArtworks";
import getArtworks from "../requests/getArtworks";

const App = () => {
  const [artworks, setArtworks] = useState();
  const [alert, setAlert] = useState(
    "If the artworks haven't loaded after a while there may be a problem with the server",
  );

  useEffect(() => {
    getArtworks(setArtworks, setAlert);
  }, []);

  return (
    <div className="App">
      <img src={certagsLogo} alt="Certags Logo" className="App__certags-logo" />
      <NavBar />
      <Routes>
        <Route
          path="/"
          element={<AllArtworks artworks={artworks} alert={alert} />}
        />
        <Route path="add-artwork" />
      </Routes>
    </div>
  );
};

export default App;
