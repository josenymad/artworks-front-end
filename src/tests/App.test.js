import React from "react";
import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import App from "../components/App";

it("renders Certags", () => {
  render(
    <BrowserRouter>
      <App />
    </BrowserRouter>,
  );
  const certags = screen.getByAltText("Certags Logo");

  expect(certags).toBeInTheDocument();
});

it("renders correctly", () => {
  const { asFragment } = render(
    <BrowserRouter>
      <App />
    </BrowserRouter>,
  );

  expect(asFragment()).toMatchSnapshot();
});
