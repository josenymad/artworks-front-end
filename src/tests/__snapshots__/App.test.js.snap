// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <img
      alt="Certags Logo"
      class="App__certags-logo"
      src="certags-web.png"
    />
    <nav
      class="navbar"
    >
      <ul
        class="navbar__list"
      >
        <li
          class="navbar__list-item"
        >
          <a
            class="navbar__list__link"
            href="/"
          >
            All Artworks
          </a>
        </li>
        <li
          class="navbar__list-item"
        >
          <a
            class="navbar__list__link"
            href="/add-artwork"
          >
            Add Artwork
          </a>
        </li>
        <li
          class="navbar__list-item"
        >
          <form
            class="navbar__list__search"
          >
            <label
              for="search"
            />
            <input
              class="navbar__list__search-text"
              id="search"
              placeholder="Company or Product code"
              type="search"
              value=""
            />
            <label
              for="search-button"
            />
            <button
              class="navbar__list__search-button"
              id="search-button"
              type="submit"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-magnifying-glass "
                data-icon="magnifying-glass"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </form>
        </li>
        <li
          class="navbar__list-item"
        >
          <button
            class="navbar__list__log-out"
            type="button"
          >
            Log Out
          </button>
        </li>
      </ul>
    </nav>
    <div
      class="all-artworks__error"
    >
      <p>
        If the artworks haven't loaded after a while there may be a problem with the server
      </p>
    </div>
  </div>
</DocumentFragment>
`;
